Project2015To2017.Core/Writing/ProjectWriter.cs

```diff
@@ -14,30 +14,33 @@ public sealed class ProjectWriter
 		private readonly ILogger logger;
 		private readonly Action<FileSystemInfo> deleteFileOperation;
 		private readonly Action<FileSystemInfo> checkoutOperation;
+		private readonly bool makeBackups;
+
 
 		public ProjectWriter(ILogger logger = null)
-			: this(logger, _ => { }, _ => { })
+			: this(logger, new ProjectWriteOptions {DeleteFileOperation = _ => { }})
 		{
 		}
 
-		public ProjectWriter(Action<FileSystemInfo> deleteFileOperation, Action<FileSystemInfo> checkoutOperation)
-			: this(null, deleteFileOperation, checkoutOperation)
+		public ProjectWriter(ProjectWriteOptions options)
+			: this(null, options)
 		{
 
 		}
 
-		public ProjectWriter(ILogger logger, Action<FileSystemInfo> deleteFileOperation, Action<FileSystemInfo> checkoutOperation)
+		public ProjectWriter(ILogger logger, ProjectWriteOptions options)
```

Comment:
We should retain old API for at least one major version. Please, add old constructor signatures that call the new APIs. A good idea would be also to place `[Obsolete]` on them.